### PR TITLE
[1921] Simplify the programmatic creation of view models

### LINKED ISSUE #1921: Simplify the programmatic creation of view models
In a project `sirius-components-view-builder`, depending pretty much only on `sirius-components-view`, we should provide utility classes to help the creation of view based models. You can have a look at the content of `org.eclipse.sirius.web.sample.papaya.view` to see some of the needs of such API.

A detailed ADR will be necessary to gather some feedback first.

### LINKED ISSUE #1960: Add a new project sirius-components-view-builder
Add a project sirius-components-view-builder with :

- IdBuilderMap
- IDiagramElementDescriptionProvider
- IEdgeDescriptionProvider
- INodeDescriptionProvider
- INodeToolProvider
- IRepresentationDescriptionProvider

### PR DESCRIPTION
**Pull request by mcharfadi**
Bug: [Simplify the programmatic creation of view models](https://github.com/eclipse-sirius/sirius-components/issues/1921)

# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [ ] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/IdBuilderMap.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
  *
  * @author sbegaudeau
  */
-public class PapayaViewCache {
+public class IdBuilderMap {
```
Comment (sbegaudeau) [inline packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/IdBuilderMap.java]: What is an IdBuilderMap supposed to be? It's not a map and it's not related to an "id".

We must explain its role in more details. It is used for two different roles:

- Store most objects that have been created but that may not have been added yet in the containment tree of the view
- Retrieve objects from were we have stored them using some predicate

Specifiers will have to manipulate the second role but not the first one and we don't need them to manipulate the implementation directly so we could provide those two interfaces:

```
/**
 * Used to store objects created by the various providers.
 *
 * @author sbegaudeau
 */
public interface IViewObjectCache {
    void put(EObject eObject)
}
```

This first interface is an internal one specifically for us.

```
/**
 * Used to retrieve diagram description elements created by providers.
 *
 * @author sbegaudeau 
 */
public interface IViewDiagramElementFinder {
    Optional<NodeDescription> getNodeDescription(String name);
    Optional<EdgeDescription> getEdgeDescription(String name);
}
```
We will have additional methods here later but this interface is for the specifiers which would not have access to the first method and would thus not be able to populate the cache by themselves. We would know what we have here and how the cache is populated.

And we would thus have something like this for the specifiers:

```
public class ViewDiagramElementFinder implements IViewObjectCache, IViewDiagramElementFinder {
    ....
}
```

Do not forget to update the JavaDoc (it is still mentioning Papaya)

**Review by sbegaudeau — CHANGES_REQUESTED** [review]

**Review by sbegaudeau — CHANGES_REQUESTED** [review]
Could you rebase this pull request? Thanks

File: packages/sirius-web/backend/sirius-web-sample-application/pom.xml [hunk 2]
```diff
@@ -202,6 +207,11 @@
 			<version>2023.4.0</version>
 			<scope>test</scope>
 		</dependency>
+		<dependency>
+			<groupId>org.eclipse.sirius</groupId>
+			<artifactId>sirius-components-view-builder</artifactId>
+			<version>2023.4.0</version>
+		</dependency>
```
Comment (gcoutable) [inline packages/sirius-web/backend/sirius-web-sample-application/pom.xml]: Duplicate

File: packages/view/backend/sirius-components-view-builder/.classpath [hunk 3]
```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (gcoutable) [inline packages/view/backend/sirius-components-view-builder/.classpath:1]: Add `src/test/java`. If this project does not have tests add a `.gitkeep` file.
But for this project may be the basic Arch Unit tests dan be added.

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+            try {
+                gen.doGen(model);
+            } catch (IOException e1) {
+                LOGGER.error(e1.getMessage());
+            }
+        });
+        LOGGER.info("Generated!");
+    }
+
+    public void doGen(GenModel model) throws IOException {
```
Comment (gcoutable) [inline packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java]: Checkstyle error here

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/ViewBuilders.java [hunk 5]
```diff
@@ -0,0 +1,657 @@
+package org.eclipse.sirius.components.view.builder.generated;
```
Comment (gcoutable) [inline packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/ViewBuilders.java:13]: It needs a copyright

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/ViewBuilders.java [hunk 6]
```diff
@@ -0,0 +1,657 @@
...
+/**
+ * Use to instantiate a new builder.
+ * @generated
+ */
+@SuppressWarnings("checkstyle:JavadocType")
+public class ViewBuilders {
+           /**
```
Comment (gcoutable) [inline packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/ViewBuilders.java]: detail : to much spaces

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/WidgetDescriptionBuilder.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
+package org.eclipse.sirius.components.view.builder.generated;
```
Comment (gcoutable) [inline packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/WidgetDescriptionBuilder.java:13]: copyright

File: CHANGELOG.adoc [hunk 8]
```diff
@@ -63,6 +63,7 @@ The filter button inside the filter bar allows to filter (hide) all visible tree
 image:doc/screenshots/filterBarFilterButton.png[Filter Bar Filter Button,30%,30%]
 
 - https://github.com/eclipse-sirius/sirius-components/issues/1830[#1830] [layout] This feature is experimental and can be activated on a diagram by adding "__EXPERIMENTAL" to its name. The new algorithm does the minimum possible to place node without overlap.
+- https://github.com/eclipse-sirius/sirius-components/issues/1921[#1921] Introducing a new project sirius-components-view-builder to create an API for view creation
```
Comment (gcoutable) [inline CHANGELOG.adoc]: Does the generator builder mean to be used by anyone?
How are we supposed to use the generator builder?

File: packages/view/backend/pom.xml [hunk 9]
```diff
@@ -15,6 +15,13 @@
 	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
 	<modelVersion>4.0.0</modelVersion>
 
+	<parent>
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>3.0.6</version>
+		<relativePath /> <!-- lookup parent from repository -->
+	</parent>
```
Comment (gcoutable) [inline packages/view/backend/pom.xml]: Why is this needed and why `relativePath`?

File: packages/view/backend/sirius-components-view-builder/pom.xml [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+                    <cleanupDaemonThreads>false</cleanupDaemonThreads>
+                    <property name="main.parentdir" location="./.." />
+                    <arguments>
+						<argument>${project.basedir}/src/main/resources/emf-merge.xml</argument>
+	                    <argument>${project.parent.basedir}/sirius-components-view/src/main/resources/model/view.genmodel</argument>
+	                	<argument>${project.basedir}/src/main/java/org/eclipse/sirius/components/view/builder/generated</argument>
+	                	<argument>org.eclipse.sirius.components.view.builder.generated</argument>
+                    </arguments>
+                </configuration>
+		</plugin>
```
Comment (gcoutable) [inline packages/view/backend/sirius-components-view-builder/pom.xml]: whitespaces are not consistent

File: packages/view/backend/sirius-components-view-builder/pom.xml [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+				<executions>
+					<execution>
+						<id>validate</id>
+						<phase>validate</phase>
+						<goals>
+							<goal>check</goal>
+						</goals>
+					</execution>
+				</executions>
+			</plugin>-->
```
Comment (gcoutable) [inline packages/view/backend/sirius-components-view-builder/pom.xml]: If not needed, remove

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java [hunk 12]
```diff
@@ -0,0 +1,304 @@
...
+                                    return new #builderClassName();
+                                }
+
+                            """.replace(BUILDER_CLASSNAME, this.builderClasssName(clazz))
+                            .replace("#className", clazz.getName()));
+                }
+            }
+
+            factory.append("""
+                    package #package;
```
Comment (gcoutable) [inline packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java:234]: Add a copyright section here

File: packages/sirius-web/backend/sirius-web-sample-application/pom.xml (hunk 2 above)
Comment (sbegaudeau) [inline packages/sirius-web/backend/sirius-web-sample-application/pom.xml]: - You are importing the same dependency twice
- We are in 2023.4.1 now

File: CHANGELOG.adoc (hunk 8 above)
Comment (sbegaudeau) [inline CHANGELOG.adoc]: This commit does more than that now. You need to detail the work done on the builder

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java [hunk 13]
```diff
@@ -0,0 +1,304 @@
...
+
+                StringBuffer body = new StringBuffer();
+
+                if (!clazz.isAbstract()) {
+                    body.append("""
+                                /**
+                                * Create instance #eObjType.
+                                * @generated
+                                */
+                                private #eObjType beingBuilt = #packageFactory.create#eObjName();
```
Comment (gcoutable) [inline packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java]: Is it possible to replace `beingBuilt` with the first lowercase type name, `barChartDescription`, `colorPalette`, and so on?

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java [hunk 14]
```diff
@@ -0,0 +1,304 @@
...
+
+                                /**
+                                * Return instance #eObjType.
+                                * @generated
+                                */
+                                protected #eObjType getBeingBuilt() {
```
Comment (gcoutable) [inline packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java]: Is it possible to rename `getBeingBuilt()` by `getBarChartDescription()`, `getColorPalette()` and so on?

File: packages/sirius-web/backend/sirius-web-sample-application/src/main/java/org/eclipse/sirius/web/sample/papaya/view/classdiagram/CDCreateClassNodeToolProvider.java [hunk 15]
```diff
@@ -45,15 +45,16 @@ public NodeTool create(PapayaViewCache cache) {
...
-        var createView = ViewFactory.eINSTANCE.createCreateView();
-        createView.setParentViewExpression("aql:selectedNode");
-        createView.setSemanticElementExpression("aql:newInstance");
-        createView.setElementDescription(classNodeDescription);
-
-        nodeTool.getBody().add(createInstance);
-        createInstance.getChildren().add(changeContext);
-        changeContext.getChildren().add(setValue);
-        setValue.getChildren().add(createView);
+        if (classNodeDescription.isPresent()) {
```
Comment (sbegaudeau) [inline packages/sirius-web/backend/sirius-web-sample-application/src/main/java/org/eclipse/sirius/web/sample/papaya/view/classdiagram/CDCreateClassNodeToolProvider.java]: Small nitpick but it would be better to rename variables like this one to `optionalXxx` to match our naming conventions and indicate to everybody the type of the variable

File: packages/view/backend/pom.xml [hunk 16]
```diff
@@ -15,6 +15,13 @@
...
+
```
Comment (sbegaudeau) [inline packages/view/backend/pom.xml]: This does not need to depend on Spring boot

File: packages/view/backend/sirius-components-view-builder/pom.xml [hunk 17]
```diff
@@ -0,0 +1,124 @@
...
+	<parent>
+		<groupId>org.eclipse.sirius</groupId>
+		<artifactId>sirius-components-view-parent</artifactId>
+		<version>2023.4.0</version>
+	</parent>
+	
+  <artifactId>sirius-components-view-builder</artifactId>
+  <name>sirius-components-view-builder</name>
+  <description>Sirius View Builder</description>
+
```
Comment (sbegaudeau) [inline packages/view/backend/sirius-components-view-builder/pom.xml]: There are lots of formatting issues in this file with spaces and tabs mixed

File: packages/view/backend/sirius-components-view-builder/pom.xml [hunk 18]
```diff
@@ -0,0 +1,124 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>org.eclipse.sirius</groupId>
+		<artifactId>sirius-components-view-parent</artifactId>
+		<version>2023.4.0</version>
```
Comment (sbegaudeau) [inline packages/view/backend/sirius-components-view-builder/pom.xml]: When this will be rebased, you'll need to consider that we are in 2023.4.1 now

File: packages/view/backend/sirius-components-view-builder/pom.xml (hunk 11 above)
Comment (sbegaudeau) [inline packages/view/backend/sirius-components-view-builder/pom.xml]: Checkstyle is required and checkstyle errors must be fixed

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/IViewDiagramElementFinder.java [hunk 19]
```diff
@@ -0,0 +1,16 @@
+package org.eclipse.sirius.components.view.builder;
```
Comment (sbegaudeau) [inline packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/IViewDiagramElementFinder.java:13]: - Missing copyright on tons of files
- Add a CodingRulesTests and ArchitectureConstants as architectural test in order to ensure that the code in this project matches our expectations

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/BarChartDescriptionBuilder.java [hunk 20]
```diff
@@ -0,0 +1,118 @@
+package org.eclipse.sirius.components.view.builder.generated;
+
+/**
+* Builder for BarChartDescriptionBuilder.
+* @generated
+*/
+@SuppressWarnings("checkstyle:JavadocType")
```
Comment (sbegaudeau) [inline packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/BarChartDescriptionBuilder.java:19]: Fix the generator and the generated code instead

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java [hunk 21]
```diff
@@ -0,0 +1,304 @@
...
+    private static final Logger LOGGER = LoggerFactory.getLogger(BuilderGenerator.class);
+
+    private String pathToJMergeFile;
+
+    private String outputDirectory;
+
+    private String basePackage = "fr.obeo.dsl.biblio.app.representationsbuilders.builders.test";
+
+    public BuilderGenerator(String pathToJMergeFile, String outputDirectory, String basePackage) {
+        super();
```
Comment (sbegaudeau) [inline packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java]: Useless super

File: packages/view/backend/sirius-components-view/src/main/resources/model/view.genmodel [hunk 22]
```diff
@@ -5,7 +5,7 @@
     modelDirectory="/sirius-components-view/src/main/java" creationIcons="false" editDirectory="/sirius-components-view-edit/src/main/java"
     editorDirectory="" modelPluginID="sirius-components-view" modelName="View" rootExtendsClass="org.eclipse.emf.ecore.impl.MinimalEObjectImpl$Container"
     codeFormatting="true" commentFormatting="true" importerID="org.eclipse.emf.importer.ecore"
-    bundleManifest="false" complianceLevel="17.0" copyrightFields="false" editPluginID="sirius-components-view-edit"
+    bundleManifest="false" copyrightFields="false" editPluginID="sirius-components-view-edit"
```
Comment (sbegaudeau) [inline packages/view/backend/sirius-components-view/src/main/resources/model/view.genmodel]: Why remove the compliance to Java 17?

**Review by sbegaudeau — DISMISSED** [review]

File: packages/view/backend/sirius-components-view/src/main/resources/model/view.genmodel (hunk 22 above)
Comment (mcharfadi) [reply packages/view/backend/sirius-components-view/src/main/resources/model/view.genmodel]: I had an issue with loading the .genmodel it crashed with this property set to 17.0

File: packages/sirius-web/backend/sirius-web-sample-application/pom.xml (hunk 2 above)
Comment (mcharfadi) [reply packages/sirius-web/backend/sirius-web-sample-application/pom.xml]: I'll rebase the PR.

File: packages/view/backend/pom.xml (hunk 16 above)
Comment (mcharfadi) [reply packages/view/backend/pom.xml]: I needed that to get slf4j dependency but I can remove logs message in BuilderGenerator

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/IViewDiagramElementFinder.java (hunk 19 above)
Comment (mcharfadi) [reply packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/IViewDiagramElementFinder.java:13]: Should I put copyright in the generated files too ?

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/BarChartDescriptionBuilder.java (hunk 20 above)
Comment (mcharfadi) [reply packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/BarChartDescriptionBuilder.java:19]: This is for the author rule but since they are generated I did no think it was worth having an author

**Review by gcoutable — CHANGES_REQUESTED** [review]

**Comment by gcoutable** [comment]
Well! @sbegaudeau also made a review, sorry for the duplicate comments.

File: packages/view/backend/pom.xml (hunk 9 above)
Comment (mcharfadi) [reply packages/view/backend/pom.xml]: This is needed for 
${project.parent.basedir}/sirius-components-view/src/main/resources/model/view.genmodel</argument> 
in the view-builder pom

File: CHANGELOG.adoc [hunk 23]
```diff
@@ -75,6 +75,11 @@ See link:how-to/contribute-custom-widget.adoc[the documentation] for more detail
 - https://github.com/eclipse-sirius/sirius-components/issues/1830[#1830] [layout] This feature is experimental and can be activated on a diagram by adding "__EXPERIMENTAL" to its name. The new algorithm does the minimum possible to place node without overlap.
 - https://github.com/eclipse-sirius/sirius-components/issues/1985[|#1985] [sirius-web] It is now possible to use in-memory View-based representations by registering them in the new `InMemoryViewRegistry`.
 These representations can be created programmatically or loaded from `.view` EMF models on startup, and do not need to be stored as documents inside a project in the database.
+- https://github.com/eclipse-sirius/sirius-components/issues/1921[#1921] [view] Added a project sirius-components-view-builder.
++
+Introducing providers interfaces to help creating view programmatically.
++
+Introducing a generator of builders aimed to help creating view programmatically, the generation makes use of emf-merge and modifications to these builders can be annoted to live during futur regeneration.
```
Comment (gcoutable) [inline CHANGELOG.adoc]: typo: annotated

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java [hunk 24]
```diff
@@ -0,0 +1,364 @@
...
+            if (!feat.getEcoreFeature().isDerived() && feat.getEcoreFeature().isChangeable()) {
+                if (!feat.getEcoreFeature().isMany()) {
+                    body.append("""
+                               /**
+                               * Setter for #accessor.
+                               * #javadoc
+                               * @generated
+                               */
+                                public #builderClassName #featName(#paramType value) {
+                                    get#eObjName().set#accessor(value);
```
Comment (gcoutable) [inline packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java]: this.get...

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java [hunk 25]
```diff
@@ -0,0 +1,364 @@
...
+                } else {
+                    body.append("""
+                                /**
+                                 * Setter for #accessor.
+                                 * #javadoc
+                                 * @generated
+                                 */
+                                public #builderClassName #featName(#paramType ... values) {
+                                    for (#paramType value : values) {
+                                        get#eObjName().get#accessor().add(value);
```
Comment (gcoutable) [inline packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java]: this.get...

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java [hunk 26]
```diff
@@ -0,0 +1,364 @@
...
+                                    get#eObjName().set#accessor(value);
+                                    return this;
+                                }
+                            """.replace(BUILDER_CLASSNAME, this.builderClasssName(clazz)).replace("#paramType", feat.getListItemType(clazz)).replace("#accessor", feat.getAccessorName())
+                            .replace("#featName", feat.getSafeName()).replace("#javadoc", this.getDocumentation(feat)).replace(BUILDER_EOBJECT_NAME, clazz.capName(clazz.getSafeUncapName())));
+                } else {
+                    body.append("""
+                                /**
+                                 * Setter for #accessor.
+                                 * #javadoc
```
Comment (gcoutable) [inline packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java]: Remove the whitespace when there is no javadoc

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java [hunk 27]
```diff
@@ -0,0 +1,364 @@
...
+        }
+
+        for (GenFeature feat : clazz.getAllGenFeatures()) {
+
+            if (!feat.getEcoreFeature().isDerived() && feat.getEcoreFeature().isChangeable()) {
+                if (!feat.getEcoreFeature().isMany()) {
+                    body.append("""
+                               /**
+                               * Setter for #accessor.
+                               * #javadoc
```
Comment (gcoutable) [inline packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java]: Remove the whitespace when there is no javadoc

**Review by gcoutable — CHANGES_REQUESTED** [review]

**Comment by gcoutable** [comment]
I take it from here

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java (hunk 27 above)
Comment (gcoutable) [reply packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java]: Done

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java (hunk 26 above)
Comment (gcoutable) [reply packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java]: Done

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java (hunk 25 above)
Comment (gcoutable) [reply packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java]: Done

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java (hunk 24 above)
Comment (gcoutable) [reply packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java]: Done

File: CHANGELOG.adoc (hunk 23 above)
Comment (gcoutable) [reply CHANGELOG.adoc]: Done

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java (hunk 27 above)
Comment (mcharfadi) [reply packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generator/BuilderGenerator.java]: Thanks for the review

File: packages/view/backend/pom.xml (hunk 9 above)
Comment (gcoutable) [reply packages/view/backend/pom.xml]: It has been removed

File: packages/view/backend/sirius-components-view/src/main/resources/model/view.genmodel (hunk 22 above)
Comment (gcoutable) [reply packages/view/backend/sirius-components-view/src/main/resources/model/view.genmodel]: I change the version of `org.eclipse.emf.codegen.ecore` to 2.25.0 in `sirius-components-view-builder` and it looks fine now

File: packages/view/backend/pom.xml (hunk 16 above)
Comment (gcoutable) [reply packages/view/backend/pom.xml]: It has been removed

File: packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/BarChartDescriptionBuilder.java (hunk 20 above)
Comment (gcoutable) [reply packages/view/backend/sirius-components-view-builder/src/main/java/org/eclipse/sirius/components/view/builder/generated/BarChartDescriptionBuilder.java:19]: IMO, unless we do not want checkstyle running on _*.generated_ package at all, keeping the _@SuppressWarnings_ is the best solution.

**Review by gcoutable — APPROVED** [review]
